File: weable/cloud/Polaris.Business/Models/Session.cs

using System.Text.Json.Serialization;
using Polaris.Business.Models.Public;

namespace Polaris.Business.Models
{
    public class SessionModel {
        [JsonPropertyName("account")]
        public PBAccountModel Account { get; set; } = new();
        
        [JsonPropertyName("token")]
        public string Token { get; set; } = "";
        
        [JsonPropertyName("name")]
        public string Name { get; set; } = "";
        
        [JsonPropertyName("domain")]
        public string Domain { get; set; } = "";
    }
}

File: electron/src/console/providers/session.ts
import {atom} from 'recoil'
import {SessionModel} from "@pnnh/polaris-business";

const sessionAtom = atom<SessionModel>({
    key: 'session',
    default: {
        account: {
            uid: '',
            username: '',
            create_time: '',
            update_time: '',
            image: '',
            description: '',
            mail: '',
            nickname: '',
            photo: '',
            urn: '',
            role: '',
        },
        token: '',
        name: '',
        domain: '',
    }
})

export {sessionAtom}
